packages/quantic/force-app/main/default/lwc/quanticRecentResultsList/__tests__/quanticRecentResultList.test.js

```diff
@@ -0,0 +1,474 @@
+          results: [],
+        };
+      });
+
+      it('should display an empty list when it is set to false', async () => {
+        const element = createTestComponent({
+          ...defaultOptions,
+          hideWhenEmpty: false,
+        });
+        updateRecentResultsState();
+        await flushPromises();
+
+        const recentResultsContainer = element.shadowRoot.querySelector(
+          selectors.quanticCardContainer
+        );
+        expect(recentResultsContainer).not.toBeNull();
+        expect(recentResultsContainer.title).toEqual(defaultRecentResultsTitle);
+
+        const recentResultsListItems = element.shadowRoot.querySelectorAll(
+          selectors.recentResultItem
+        );
+        expect(recentResultsListItems.length).toEqual(0);
+      });
+
+      it('should not display a recent results card when it is set to true', async () => {
+        const element = createTestComponent({
+          ...defaultOptions,
+          hideWhenEmpty: true,
+        });
+        await flushPromises();
+
+        const recentResultsContainer = element.shadowRoot.querySelector(
+          selectors.quanticCardContainer
+        );
+        expect(recentResultsContainer).toBeNull();
+      });
+    });
+
+    describe('with a custom #target value', () => {
+      it('should pass the target property to the result link component', async () => {
+        const customTarget = '_blank';
+        const exampleResults = mockResults;
+        recentResultsListState = {
+          ...recentResultsListState,
+          results: exampleResults,
+        };
+        const element = createTestComponent({
+          ...defaultOptions,
+          target: customTarget,
+        });
+        await flushPromises();
+
+        const recentResultLinks = element.shadowRoot.querySelectorAll(
+          selectors.recentResultItem
+        );
+
+        recentResultLinks.forEach((link) => {
+          expect(link.getAttribute('target')).toEqual(customTarget);
+        });
+      });
+    });
+
+    describe('when the #isCollapsed property is false', () => {
+      it('should display the card title, the collapse button icon and the recent results list', async () => {
+        const collapseRecentResultsLabel = 'Collapse Recent results';
+        const collapseRecentResultsIcon = 'utility:dash';
+        const element = createTestComponent({
+          ...defaultOptions,
+          isCollapsed: false,
+        });
+        updateRecentResultsState();
+        await flushPromises();
+
+        const recentResultsContainer = element.shadowRoot.querySelector(
+          selectors.quanticCardContainer
+        );
+        expect(recentResultsContainer).not.toBeNull();
+        expect(recentResultsContainer.title).toEqual(defaultRecentResultsTitle);
+
+        const toggleButton = element.shadowRoot.querySelector(
+          selectors.toggleButton
+        );
+
+        expect(toggleButton).not.toBeNull();
+        expect(toggleButton.alternativeText).toEqual(
```

Comment:
`toEqual` is to deep compare objects.
please pay attention to that in all your tests.